README.rst

```diff
@@ -87,13 +87,9 @@ Advanced customisation
 Getting IP
 ~~~~~~~~~~
 
-If you want to have a custom behaviour when getting IP, you can create a
-custom function that takes request as a parameter and specify path to it
-in the ``BASIC_AUTH_GET_CLIENT_IP_FUNCTION`` settings, e.g.
-
-.. code:: python
-
-   BASIC_AUTH_GET_CLIENT_IP_FUNCTION = 'utils.ip.get_client_ip'
+Under the hood, `django-ipware <https://github.com/un33k/django-ipware>`_ is used
+for detecting the client's IP address. See their documentation for customization and
+security hardening.
```

Comment:
**Question:** Should we add an example or explanation of implications for users?

This is a massively breaking change with security implications and it'll be hard to communicate to people. They may not spot it straight away as this is very often not used in local development.

We might want to enforce that users set this up explicitly? Or should we abstract getting client IP and still let users choose between the legacy mechanism and ipware?

I think the steps are:
1. Move `baipw.utils.get_client_ip` to `baipw.ip.legacy_get_client_ip`.
2. Create ipware function call in `baipw.ip.get_ipware_client_ip`.
3. Add deprecation warning for the function (add a new issue that I'll handle before release).
4. Request the setting to be set somewhere. Probably a Django check so it does not pass tests and does not deploy
   ```python
   # somewhere else
   if not settings.BASIC_AUTH_GET_CLIENT_IP_FUNCTION:
      raise GetClientIpFunctionNotSet
   ```

This would look like this in settings:

```
#  settings.py
BASIC_AUTH_GET_CLIENT_IP_FUNCTION = 'baipw.ip.legacy_get_client_ip'
## or....
BASIC_AUTH_GET_CLIENT_IP_FUNCTION = 'baipw.ip.get_ipware_client_ip'
```

Sorry for making this more complicated. I am a tad terrified about the prospect of smuggling in big changes like that that may affect security of library users.